lab2.js

```diff
@@ -85,13 +112,29 @@ var hello = {
 // speak, and method (that you'll place on the prototype) called
 // sayHello.
 
-function SentientBeing () {
+function SentientBeing(homePlanet, language) {
+  this.homePlanet = homePlanet;
+  this.language = language;
   // TODO: specify a home planet and a language
   // you'll need to add parameters to this constructor
 }
 
 // sb is a SentientBeing object
-function sayHello (sb) {
+
+// Many thanks to Michael and Shawn for posting their code to GitHub and giving me the clue I needed to get the syntax correct!
+SentientBeing.prototype.sayHello = function(sb) {
+  if (sb.language === 'klingon') {
+    console.log(hello[this.language]);
+    return hello.klingon;
+  }
+  if (sb.language == 'romulan') {
+    console.log(hello[this.language]);
+    return hello.romulan;
+  }
+  if (sb.language == 'federation standard') {
+    console.log(hello[this.language]);
+    return hello['federation standard'];
```

Comment:
this is a good use case for bracket notation. If the languages on the sentient being match what's being used for parameters in the hello variable, you can do:
console.log(hello[this.language]);
return hello[sb.language];